src/client/common/utils/async.ts

```diff
@@ -237,3 +237,16 @@ export async function flattenIterator<T>(iterator: AsyncIterator<T, void>): Prom
     }
     return results;
 }
+
+export class ChainedExecutions<T> {
+    private pendingCellExecution?: Promise<T> | undefined;
+    public async chainExecution(next: () => Promise<T>): Promise<T> {
+        if (this.pendingCellExecution) {
+            await this.pendingCellExecution;
```

Comment:
I believe this may screw stuff up if there's an error. The error will continually come out.

I think you should clear pendingCellExecution before awaiting it. I think we can await it only once.

Reply:
You can test this by removing ipykernel, run a cell, then attempt to run again.

Reply:
Yup missed that, thanks